universal-application-tool-0.0.1/app/auth/ProfileFactory.java

```diff
@@ -1,7 +1,22 @@
 package auth;
 
+import com.google.common.base.Preconditions;
+import java.time.Clock;
+import javax.inject.Inject;
+
 public class ProfileFactory {
+
```

Comment:
question: Is it worth adding unit tests for ProfileFactory now that it does a bit more?

Reply:
imo `BrowserTest.noCredLogin()` does it - I probably couldn't write a test that ran much faster that was worth running.